share/wake/lib/core/json.wake

```diff
@@ -538,3 +560,359 @@ export def lastValueInList (values: List JValue): Result JValue Error =
             | flatten
             | deduplicateJObjectKeys lastValueInList
             | rmap JObject
+
+# If the input is a string, returns a singelton list
+# containing that string. This can be used to extract
+# a string from a JValue or in list operation chains
+# as a safe JValue to String conversion.
+#
+# Examples:
+# ```
+#   jString (JString "foo") = "foo", Nil
+#   jString (JInteger 10) = Nil
+#   single json |> jField "name" |> jString = "Alice", Nil
+#   single json |> jField "age" |> jString = Nil
+# ```
+export def jString: JValue => List String = match _
+    JString s = single s
+    _         = Nil
+
+# TODO: Fix this doc
+# If `input` is a JString, updates the string using `editFn`.
+#
+# Parameters:
+#   - `editFn`: The function to update a string with
+#   - `input`: the JValue to be modified
+#
+# Examples:
+# ```
+#   editJString ("{_}man") (JString "bat") = JString "batman"
+#   editJString ("{_}man") (JInteger 10) = JInteger 10
+#   (editJField "name" $ editJString ("I'm {_}")) batman = JObject ("name" :-> JString "I'm batman", Nil)
+#   def rename (newName: String): (person: JValue) => JValue = editJField "name" $ editJString (\_ newName)
+# ```
+export def editJString (editFn: String => Result String Error): (input: JValue) => Result JValue Error = match _
+    JString s =
+        require Pass es = editFn s
+        Pass (JString es)
+    _         = failWithError "attempt to use editJString on non-JString"
```

Comment:
While this is a very verbose programmer-facing explanation, I think the error message we want is user-facing.

Something along the lines of:
"/tree/person[3]/name: not a string"